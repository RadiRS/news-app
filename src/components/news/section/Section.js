import React, { Component } from 'react';
import styled from 'styled-components';

export class Section extends Component {
  render() {
    return (
      <Container>
        <Content>
          <Title>Test</Title>
          <Author>Test</Author>
          <Date>Test</Date>
        </Content>
        <Image source={{ uri: 'http://lorempixel.com/640/480' }} />
      </Container>
    );
  }
}

export default Section;

const Container = styled.View``;

const Content = styled.View``;

const Image = styled.Image``;

const Title = styled.Text``;

const Author = styled.Text``;

const Date = styled.Text``;
