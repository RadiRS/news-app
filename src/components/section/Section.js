import React from 'react';
import styled from 'styled-components';
import { Fonts } from '../../themes';

const Section = props => {
  const { text } = props;

  return (
    <Container onLongPress={() => alert(text)}>
      <Text>{text}</Text>
    </Container>
  );
};

const Container = styled.TouchableOpacity`
  flex-direction: row;
  align-items: center;
  background: white;
  height: 60px;
  padding: 12px;
  margin: 8px;
  border-radius: 10px;
  elevation: 5;
`;

const Text = styled.Text`
  font-family: ${Fonts.type.bold};
  font-size: 16px;
  font-weight: ${Fonts.weight.large};
`;

export default Section;
