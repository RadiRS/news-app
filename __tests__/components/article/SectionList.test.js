import 'react-native';
import 'jest-styled-components';
import React from 'react';
import renderer from 'react-test-renderer';
import Section from '../../../src/components/section';

test('renders Section component correctly', () => {
  const tree = renderer.create(<Section text="" />).toJSON();
  expect(tree).toMatchSnapshot();
});
