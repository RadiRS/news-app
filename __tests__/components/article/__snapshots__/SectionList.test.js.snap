// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Section component correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": 10,
      "elevation": 5,
      "flexDirection": "row",
      "height": 60,
      "marginBottom": 8,
      "marginLeft": 8,
      "marginRight": 8,
      "marginTop": 8,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Domine Bold",
          "fontSize": 16,
          "fontWeight": "600",
        },
      ]
    }
  >
    
  </Text>
</View>
`;
